app/services/user_subscriptions/create.rb

```diff
@@ -0,0 +1,72 @@
+module UserSubscriptions
+  class Create
+    attr_accessor :user, :source_type, :source_id, :subscriber_email, :response
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(user, user_subscription_params)
+      @user = user
+      @source_type = user_subscription_params[:source_type]
+      @source_id = user_subscription_params[:source_id]
+      @response = new_response
+
+      # TODO: [@thepracticaldev/delightful]: uncomment this once email confirmation is re-enabled
+      # @subscriber_email = user_subscription_params[:subscriber_email]
+    end
+
+    def call
+      return response if invalid_source_type?
+
+      source = source_type.constantize.find_by(id: source_id)
+      return response if invalid_source?(source)
+
+      # TODO: [@thepracticaldev/delightful]: uncomment this once email confirmation is re-enabled
+      # return response if subscriber_email_mismatch?
+
+      user_subscription = source.build_user_subscription(user)
+      if user_subscription.save
+        response.success = true
+        response.data = user_subscription
+      else
+        response.error = user_subscription.errors.full_messages.to_sentence
```

Comment:
A while back I added a helper on `ApplicationRecord`, so this can be 

```ruby
response.error = user_subscription.errors_as_sentence
```

Reply:
Sweet - updated!